crates/trie/parallel/src/root.rs

```diff
@@ -233,6 +233,15 @@ pub enum ParallelStateRootError {
     /// Provider error.
     #[error(transparent)]
     Provider(#[from] ProviderError),
+    /// Sparse Tree not revealed .
+    #[error("{_0}")]
+    SparseTreeNotRevealed(String),
+    /// Retention not enabled .
+    #[error("{_0}")]
+    RetentionNotEnabled(String),
+    /// Sparse Tree not revealed and Retention not enabled .
+    #[error("{_0}")]
+    SparseTreeAndRetentionNotAvailable(String),
```

Comment:
```suggestion
    /// Sparse Tree not revealed and Retention not enabled .
    #[error("both root and trie update should be visible")]
    SparseTreeAndRetentionNotAvailable,
```
applies to the other new variants too